### PR TITLE
Day9-Ex2-FeedActivity

### PR DESCRIPTION
**Pull request by anhhuy173**
#### Description (Required)
- [x] create viewpager 
- [x] read SQLite
- [x] load image with Universal  
- [x] add next and previous

#### Reference (Optional)

### CODE REVIEW

File: app/src/main/java/vn/asiantech/internship/ui/adapter/FeedAdapter.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    public int getItemCount() {
+        return mFoodLists.size();
+    }
+
+    /**
+     * View Holder for Feed Item
+     * Created by huypham on 15/6/2017
+     */
+
+    private class ViewHolder extends RecyclerView.ViewHolder {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/ui/adapter/FeedAdapter.java]: ERROR: Class ViewHolder should be declared as final.

File: app/build.gradle [hunk 2]
```diff
@@ -26,5 +32,10 @@ dependencies {
     })
     compile 'com.android.support:appcompat-v7:25.3.1'
     compile 'com.android.support.constraint:constraint-layout:1.0.2'
+    compile 'com.android.support:recyclerview-v7:25.3.1'
+    compile 'com.android.support:cardview-v7:25.3.1'
+    compile 'de.hdodenhof:circleimageview:2.1.0'
     testCompile 'junit:junit:4.12'
 }
+
+apply from: "https://raw.githubusercontent.com/monstar-lab/gradle-android-ci-check/1.3.1/ci.gradle"
```
Comment (at-vinhhuynh) [inline app/build.gradle]: Add break line

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -9,13 +9,17 @@
...
+        <activity android:name=".ui.feed.FeedActivity">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN"/>
 
                 <category android:name="android.intent.category.LAUNCHER"/>
             </intent-filter>
         </activity>
+        <activity android:name=".MainActivity">
+
+        </activity>
```
Comment (at-vinhhuynh) [inline app/src/main/AndroidManifest.xml]: Add break line

File: app/src/main/java/vn/asiantech/internship/models/Food.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package vn.asiantech.internship.models;
+
+/**
+ * model food
+ * Created by huypham on 15/06/2017.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/Food.java]: Remove break line

File: app/src/main/java/vn/asiantech/internship/models/Food.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+public class Food {
+    private String foodName;
+    private int[] image;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/Food.java]: variable name to `images`

File: app/src/main/java/vn/asiantech/internship/models/Food.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    private String content;
+
+    public Food(String foodName, int[] image, String content) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/models/Food.java]: `image` to `images`

File: app/src/main/java/vn/asiantech/internship/ui/adapter/FeedAdapter.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.List;
+
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.models.Food;
+
+/**
+ * Adapter for Feed
+ * Created by huypham on 15/06/2017.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/ui/adapter/FeedAdapter.java]: remove break line

File: app/src/main/java/vn/asiantech/internship/ui/adapter/FeedAdapter.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    @Override
+    public int getItemCount() {
+        return mFoodLists.size();
+    }
+
+    /**
+     * View Holder for Feed Item
+     * Created by huypham on 15/6/2017
+     */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/ui/adapter/FeedAdapter.java]: remove break line

File: app/src/main/java/vn/asiantech/internship/ui/adapter/ImageAdapter.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import vn.asiantech.internship.R;
+
+/**
+ * Adapter for Image View Pager
+ * Created by huypham on 15/06/2017.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/ui/adapter/ImageAdapter.java]: remove break line

File: app/src/main/java/vn/asiantech/internship/ui/feed/FeedActivity.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+import android.support.v4.app.FragmentTransaction;
+import android.support.v7.app.AppCompatActivity;
+
+import vn.asiantech.internship.R;
+
+/**
+ * Activity for Feed
+ * Created by huypham on 15/06/2017.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/ui/feed/FeedActivity.java]: remove break line

File: app/src/main/java/vn/asiantech/internship/ui/feed/FeedActivity.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+public class FeedActivity extends AppCompatActivity {
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_feed);
+        setFragment();
+    }
+
+    private void setFragment() {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/ui/feed/FeedActivity.java]: change method name to `replaceFragment`

File: app/src/main/java/vn/asiantech/internship/ui/feed/FeedFragment.java [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.models.Food;
+import vn.asiantech.internship.ui.adapter.FeedAdapter;
+
+/**
+ * Fragment for Feed
+ * Created by anhhuy on 15/06/2017.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/ui/feed/FeedFragment.java]: remove break line

File: app/src/main/java/vn/asiantech/internship/ui/feed/FeedFragment.java [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+public class FeedFragment extends Fragment {
+    private RecyclerView mRecyclerViewFeed;
+    private List<Food> mFoodLists;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/ui/feed/FeedFragment.java]: variable name to `mFoodList` or `mFoods`

File: app/src/main/res/layout/item_list_feed.xml [hunk 14]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="utf-8"?>
+
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/item_list_feed.xml]: remove break line

File: app/src/main/java/vn/asiantech/internship/ui/adapters/FeedAdapter.java [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+    public int getItemCount() {
+        return mFeedList.size();
+    }
+
+    /**
+     * View Holder for Feed Item
+     * Created by huypham on 15/6/2017
+     * Change on 19/06/2017 - press on next and previous
+     */
+    private class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/ui/adapters/FeedAdapter.java]: ERROR: Class ViewHolder should be declared as final.

File: app/src/main/java/vn/asiantech/internship/ui/feed/FeedFragment.java [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+            mRecyclerViewFeed.setAdapter(feedAdapter);
+            mRecyclerViewFeed.addItemDecoration(new RecyclerView.ItemDecoration() {
+                @Override
+                public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
+                    int position = parent.getChildAdapterPosition(view);
+                    outRect.set(25, position == 0 ? 10 : 17, 25, 17);
+                }
+            });
+            itemDatabase.closeDatabase();
+        } catch (IOException e){
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/ui/feed/FeedFragment.java]: ERROR: '{' is not preceded with whitespace.

File: app/build.gradle [hunk 17]
```diff
@@ -26,5 +32,11 @@ dependencies {
...
     compile 'com.android.support:appcompat-v7:25.3.1'
     compile 'com.android.support.constraint:constraint-layout:1.0.2'
+    compile 'com.android.support:recyclerview-v7:25.3.1'
+    compile 'com.android.support:cardview-v7:25.3.1'
+    compile 'de.hdodenhof:circleimageview:2.1.0'
+    compile 'com.nostra13.universalimageloader:universal-image-loader:1.9.5'
     testCompile 'junit:junit:4.12'
 }
+
+apply from: "https://raw.githubusercontent.com/monstar-lab/gradle-android-ci-check/1.3.1/ci.gradle"
```
Comment (at-vinhhuynh) [inline app/build.gradle]: add a break line at end òf this file

File: app/src/main/java/vn/asiantech/internship/database/FeedItemDatabase.java [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.models.Feed;
+
+/**
+ * Read Database list_image
+ * Created by huypham on 20/06/2017.
+ */
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/database/FeedItemDatabase.java]: remove break line

**Comment by at-tamnguyen2** [comment]
Thanks @anhhuy173 
LGTM

**Comment by anhhuy173** [comment]
Thanks @at-tamnguyen2 and @at-vinhhuynh reviewed for me 👍

**Comment by at-tienhoang** [comment]
LGTM! thank you
LGTM

**Pull request merged**